tools/release_notes.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/python3
+#
+# 2021 Jafar Al-Gharaibeh, ATCorp
+#
+# Generate a draft FRR release notes
+#
+
+import sys
+import os
+import getopt
+import subprocess
+
+def run(cmd):
+    proc = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+    rv = proc.communicate("")[0].decode("UTF-8")
+    proc.wait()
+    return rv
+
+
+def usage(n):
+    print(os.path.basename(__file__), " [-b <branch>] [-t <tag> ]")
+    print("    Generate one line logs for non merge commits")
+    print("   -branch: branch name to use, default to HEAD")
+    print("   -tag   : generate logs up to this tag, default to latest tag")
+    sys.exit(n)
+
+
+def main(argv):
+    branch = tag = None
+    try:
+        opts, args = getopt.getopt(argv, "hb:t:", ["branch=", "tag="])
+    except getopt.GetoptError:
+        usage(2)
+    for opt, arg in opts:
+        if opt == "-h":
+            usage(0)
+        elif opt in ("-b", "--branch"):
+            branch = arg
+        elif opt in ("-t", "--tag"):
+            tag = arg
+
+    if branch is None:
+        branch = "HEAD"
+    if tag is None:
+        tag = run(["git", "describe", "--abbrev=0"]).strip("\n")
+
+    chnglog = run(
+        ["git", "log", "--no-merges", "--pretty=format:'%s%d'", tag + ".." + branch]
+    )
+    chnglog = chnglog.split("\n")
+
+    chnglist = []
+    daemons = [
```

Comment:
Would be easier to identify the daemon by looking at the modified directory's name? I mean if a commit touched `bgpd/bgp_open.c` it means it's BGP for sure. I just suggest avoiding (forgetting) adopting this file in the future.

Reply:
We could, but by the time we have a commit merged, it must be in the correct format already since it is a requirement for any commit. 

> I just suggest avoiding (forgetting) adopting this file in the future.
I don't understand this part, can you elaborate please?

Reply:
All good, I was just about like adding a new daemon to the list :)